qemu/tests/nvdimm.py

```diff
@@ -0,0 +1,141 @@
+        status, output = self.session.cmd_status_output(umount_cmd, timeout=10)
+        if status != 0:
+            raise error.TestFail("Umount nvdimm device failed: %s" % output)
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    Run nvdimm cases:
+    1) Boot guest with nvdimm device backed by a host file
+    2) Login to the guest
+    3) Check nvdimm in guest
+    4) Format nvdimm device in guest
+    5) Mount nvdimm device in guest
+    6) Create a file in the mount point in guest
+    7) Check the md5 value of the file
+    8) Umount the nvdimm device in guest
+    9) Reboot the guest
+    10) Remount nvdimm device in guest
+    11) Check if the md5 value of the nv_file changes
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    nvdimm_test = NvdimmTest(params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    error.context("Verify nvdimm in monitor", logging.info)
+    if not nvdimm_test.verify_nvdimm(vm):
+        raise error.TestFail("Verify nvdimm in monitor failed")
+    error.context("Login to the guest", logging.info)
+    login_timeout = int(params.get("login_timeout", 360))
+    nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
+    cmd_timeout = int(params.get("cmd_timeout", 360))
+    error.context("Mount nvdimm in guest", logging.info)
+    nvdimm_test.mount_nvdimm()
+    nv_file = params.get('nv_file', "/mnt/nv")
+    dd_cmd = "dd if=/dev/urandom of=%s bs=1K count=200" % nv_file
+    status, output = nvdimm_test.session.cmd_status_output(dd_cmd,
+                                                           timeout=cmd_timeout)
+    if status != 0:
+        raise error.TestFail("%s failed: %s" % (dd_cmd, output))
+
+    md5_cmd = "md5sum %s" % nv_file
+    status, output = nvdimm_test.session.cmd_status_output(md5_cmd,
+                                                           timeout=cmd_timeout)
+    if status != 0:
+        raise error.TestFail("Failed to get md5: %s" % output)
+    orig_md5 = output
+    nvdimm_test.umount_nvdimm()
+    nvdimm_test.session = vm.reboot()
+    nvdimm_test.mount_nvdimm(format_device="no")
+    status, output = nvdimm_test.session.cmd_status_output(md5_cmd,
+                                                           timeout=cmd_timeout)
+    if status != 0:
+        raise error.TestFail("Failed to get md5: %s" % output)
+    new_md5 = output
+    error.context("Compare md5", logging.info)
+    if new_md5 != orig_md5:
+        raise error.TestFail("'%s' changed. The original md5 is '%s',"
+                             " current md5 is '%s'"
+                             % (nv_file, orig_md5, new_md5))
+    nvdimm_test.umount_nvdimm()
+    nvdimm_test.session.close()
+    vm.destroy()
+    post_cmd = params.get('post_command')
+    if post_cmd:
+        logging.info("Clean the nv_backend file...")
+        utils.run(post_cmd)
```

Comment:
For those clean operations, the case would be failed if one step failed, right?
which is not expected, right?
I think those clean operations should ignore status here.